homeassistant/components/timer/__init__.py

```diff
@@ -384,7 +384,7 @@ def _async_finished(self, time):
         )
         self.async_write_ha_state()
 
-    async def async_update_config(self, config: dict) -> None:
+    async def async_update_config(self, config: ConfigType) -> None:
```

Comment:
Can `__init__` and `from_yaml` be adjusted to use `ConfigType`, to match in `timer`?